packages/block-library/src/image/image.js

```diff
@@ -1043,10 +1061,43 @@ export default function Image( {
 		);
 	}
 
+	/**
+	 * Set the post's featured image with the current image.
+	 */
+	const setPostFeatureImage = () => {
+		setFeaturedImage( id );
+	};
+
+	/**
+	 * Create a menu item to set the image as the post's featured image.
+	 *
+	 * @param {Object} props                   - Props for block settings menu control.
+	 * @param {Object} props.selectedClientIds - Selected client ids.
+	 *
+	 * @return {void} - Menu item to set the image as the post's featured image.
+	 */
```

Comment:
This component doesn't have any props, and since it's an internal component, I personally don't think it needs JSDoc at all.